core/src/main/java/org/geysermc/geyser/registry/populator/BlockRegistryPopulator.java

```diff
@@ -110,7 +111,40 @@ private static void registerBedrockBlocks() {
         var blockMappers = ImmutableMap.<ObjectIntPair<String>, Remapper>builder()
                 .put(ObjectIntPair.of("1_20_80", Bedrock_v671.CODEC.getProtocolVersion()), Conversion685_671::remapBlock)
                 .put(ObjectIntPair.of("1_21_0", Bedrock_v685.CODEC.getProtocolVersion()), Conversion712_685::remapBlock)
-                .put(ObjectIntPair.of("1_21_20", Bedrock_v712.CODEC.getProtocolVersion()), tag -> tag)
+                .put(ObjectIntPair.of("1_21_20", Bedrock_v712.CODEC.getProtocolVersion()), Conversion729_712::remapBlock)
+                .put(ObjectIntPair.of("1_21_30", Bedrock_v729.CODEC.getProtocolVersion()), tag -> { // TODO: Remove me when mappings is updated
+                    if(Objects.equals(tag.getString("name"), "minecraft:sponge")) {
+                        NbtMapBuilder builder = tag.getCompound("states").toBuilder();
+                        builder.remove("sponge_type");
+                        NbtMap states = builder.build();
+                        return tag.toBuilder().putString("name", "minecraft:sponge").putCompound("states", states).build();
+                    }
```

Comment:
1. fix spacing
2. use `"string literal".equals(x)` instead of `Object.equals(x, y)` in these kinds of cases
3. i understand this was just intended as temp code but why is the block name being replaced with the same value?